tests/test_generators.py

```diff
@@ -617,11 +617,39 @@ def test_xor_gate_strength(self):
     def test_multiplication_circuit(self):
 
         # Verify correct variables for 3x3 circuit
-        mc3_vars = ['p5', 'p3', 'p0', 'and2,2', 'and0,1', 'carry2,1', 'and0,2',
-        'p4', 'and1,0', 'p1', 'carry1,0', 'and1,1', 'sum1,1', 'carry1,1',
-        'and1,2', 'carry3,0', 'and2,0', 'p2','carry2,0', 'and2,1', 'sum2,1'].sort()
+        mc3_vars = {
+            'a0', 'a1', 'a2',
+            'b0', 'b1', 'b2',
+            'and0,1', 'and0,2', 'and1,0', 'and1,1', 'and1,2', 'and2,0', 'and2,1', 'and2,2',
+            'sum1,1', 'sum1,2',
+            'carry1,0', 'carry1,1', 'carry1,2', 'carry2,0', 'carry2,1',
+            'p0', 'p1', 'p2', 'p3', 'p4', 'p5',
+        }
         bqm = dimod.generators.multiplication_circuit(3)
-        self.assertEqual((list(bqm.variables)).sort(), mc3_vars)
```

Comment:
Good catch, though I am confused about how this was passing before?

Reply:
The definition mc3_vars had the same problem. The this was testing for None == None regardless of input so it could never fail.